audbackend/core/backend/minio.py

```diff
@@ -0,0 +1,402 @@
+        except minio.error.S3Error:
+            return False
+        return True
+
+    def _get_file(
+        self,
+        src_path: str,
+        dst_path: str,
+        verbose: bool,
+    ):
+        r"""Get file from backend."""
+        src_path = self.path(src_path)
+        src_size = self._client.stat_object(self.repository, src_path).size
+        chunk = 4 * 1024
+        with audeer.progress_bar(total=src_size, disable=not verbose) as pbar:
+            desc = audeer.format_display_message(
+                "Download {}".format(os.path.basename(str(src_path))),
+                pbar=True,
+            )
+            pbar.set_description_str(desc)
+            pbar.refresh()
+
+            dst_size = 0
+            try:
+                response = self._client.get_object(self.repository, src_path)
+                with open(dst_path, "wb") as dst_fp:
+                    while src_size > dst_size:
+                        data = response.read(chunk)
+                        n_data = len(data)
+                        if n_data > 0:
+                            dst_fp.write(data)
+                            dst_size += n_data
+                            pbar.update(n_data)
+            finally:
+                response.close()
+                response.release_conn()
+
+    def _ls(
+        self,
+        path: str,
+    ) -> typing.List[str]:
+        r"""List all files under sub-path."""
+        path = self.path(path)
+        objects = self._client.list_objects(
+            self.repository,
+            prefix=path,
+            recursive=True,
+        )
+        paths = [self._collapse(obj.object_name) for obj in objects]
+
+        return paths
+
+    def _move_file(
+        self,
+        src_path: str,
+        dst_path: str,
+        verbose: bool,
+    ):
+        r"""Move file on backend."""
+        self._copy_file(src_path, dst_path, verbose)
+        self._remove_file(src_path)
+
+    def _open(
+        self,
+    ):
+        r"""Open connection to backend."""
+        # At the moment, session management is handled automatically.
+        # If we want to manage this ourselves,
+        # we need to use the `http_client` argument of `minio.Minio`
+        if not self._client.bucket_exists(self.repository):
+            utils.raise_file_not_found_error(self.repository)
+
+    def _owner(
+        self,
+        path: str,
+    ) -> str:
+        r"""Get owner of file on backend."""
+        path = self.path(path)
+        # TODO: owner seems to be empty,
+        # need to check if we have to manage this ourselves?
+        owner = self._client.stat_object(self.repository, path).owner_name
+        return owner
+
+    def path(
+        self,
+        path: str,
+    ) -> str:
+        r"""Convert to backend path.
+
+        Args:
+            path: path on backend
+
+        Returns:
+            path
+
+        """
+        path = path.replace(self.sep, "/")
+        if path.startswith("/"):
```

Comment:
In Python versions greater than 3.8 it would be safe to remove the conditional and replace it with. `path = path.removeprefix("/")`.  I believe that it is getting close to deprecating 3.8?

Reply:
When removing Python 3.8, there are other things we should update, e.g. if I remember correctly the `typing` module might no longer be needed. I would prefer to first create a general issue to gather all the things we would like to update when removing Python 3.8, and then do this in dedicated pull requests.

Which means for now in this pull request we need to still support Python 3.8.

Reply:
This is now tracked in [tools/meta#12] internally.